vscode-dotnet-runtime-library/src/Utils/IFileUtilities.ts

```diff
@@ -8,12 +8,12 @@ import { IEventStream } from '../EventStream/EventStream';
 
 export abstract class IFileUtilities
 {
-    public abstract writeFileOntoDisk(scriptContent: string, filePath: string, eventStream : IEventStream) : void;
+    public abstract writeFileOntoDisk(scriptContent: string, filePath: string, alreadyHoldingLock : boolean, eventStream? : IEventStream) : void;
 
     /**
      * @param directoryToWipe the directory to delete all of the files in if privilege to do so exists.
      */
-    public abstract wipeDirectory(directoryToWipe : string, eventStream : IEventStream) : void;
+    public abstract wipeDirectory(directoryToWipe : string, eventStream? : IEventStream, fileExtensionsToDelete? : string[]) : void;

```

Comment:
what led to the eventStreams being nullable here? We should have some stream the entire time for logging/telemetry purposes, right?

Reply:
The `CommandExecutor` `AcquisitionContext` with the `EventStream` may be `null` in rare scenarios be null because:

The `TelemetryUtilities` needs to call the `CommandExecutor` to edit the environment, and in creating the `Executor`, needs to pass its `Context` -> The Context of the `TelemetryUtilities` may be `null`, because it needs to be created before the global event stream can be created, so that the `TelemetryUtilities` can be added as a listener to that event stream -> The `context` of `TelemetryUtilities` containing the `EventStream` is added to it later after creation, so there is a point in time where it is null between when it is created, when the event stream is created and subscribed to it, and then when the event stream is added back in to the `TelemetryUtilities`. That point 'expanded' during the refactor, where almost every class needed to take a dependency on `AcquisitionContext` to correctly post an Error, though most classes already had it as one long ago.

The commands that do not create an `AcquisitionContext` may take actions that call the `TelemetryUtilities` before it has an `AcquisitionContext` with an `EventStream`, so theoretically some of that would not be recorded if they are called before an Acquisition based command is called.

It should not be this way, I agree. Fixing the Context class is non trivial but likely doable

Reply:
TLDR: If someone changes the VS Code telemetry setting before ever running an acquisition command, then that would not show up in our logs, nor would someone enabling telemetry come up in our telemetry.

Reply:
This makes perfect sense, thank you for walking through that!